source/disassemble.h

```diff
@@ -74,13 +75,25 @@ class InstructionDisassembler {
   void SetGreen();
 
  private:
+  void ResetColor(std::ostream& stream) const;
+  void SetGrey(std::ostream& stream) const;
+  void SetBlue(std::ostream& stream) const;
+  void SetYellow(std::ostream& stream) const;
+  void SetRed(std::ostream& stream) const;
+  void SetGreen(std::ostream& stream) const;
+
   // Emits an operand for the given instruction, where the instruction
   // is at offset words from the start of the binary.
-  void EmitOperand(const spv_parsed_instruction_t& inst,
-                   const uint16_t operand_index);
+  void EmitOperand(std::ostream& stream, const spv_parsed_instruction_t& inst,
+                   const uint16_t operand_index) const;
 
   // Emits a mask expression for the given mask word of the specified type.
-  void EmitMaskOperand(const spv_operand_type_t type, const uint32_t word);
+  void EmitMaskOperand(std::ostream& stream, const spv_operand_type_t type,
+                       const uint32_t word) const;
+
+  // Generate part of the instruction as a comment to be added to
+  // |id_comments_|;
+  void GenerateFutureComment(const spv_parsed_instruction_t& inst);
```

Comment:
The name of this function is not very clear. The concept of a "future comment" is not used anywhere else. It might be better if the name was related to the member variable it write to. Something about "id comments". I'm having trouble coming up with a good name that is clear that the comment will appear next to a different instruction.

Possibilities:

`GenerateIdComments`
`GenerateFutureIdComments`
`GenerateIdCommentsForForwardReferences`

If you can come up with someelse, that is better please do.  I won't require another review for this. I trust whatever you come up with.